### PR TITLE
Add a flag to specify an artifact by hash. This is required when uplo…

### PR DESCRIPTION
**Pull request by dlorenc**
…ading by URL.

Signed-off-by: Dan Lorenc <dlorenc@google.com>

### CODE REVIEW

File: cmd/rekor-cli/app/pflags.go [hunk 1]
```diff
@@ -79,6 +79,7 @@ func addArtifactPFlags(cmd *cobra.Command) error {
 	cmd.Flags().Var(&fileOrURLFlag{}, "public-key", "path or URL to public key file")
 
 	cmd.Flags().Var(&fileOrURLFlag{}, "artifact", "path or URL to artifact file")
+	cmd.Flags().String("artifact-hash", "", "hex encdoed hash of artifact (when using URL)")
```
Comment (bobcallaway) [inline cmd/rekor-cli/app/pflags.go]: I'd change this to be of type `uuidFlag` so you get the checking for hex string of 64 char long; also maybe put that it must be SHA256 in the help text?

Comment (dlorenc) [reply cmd/rekor-cli/app/pflags.go]: Nice,done.

**Review by lukehinds — APPROVED** [review]
looks ok to me, will `https://get.jenkins.io/war-stable/2.277.3/jenkins.war` persist?

**Comment by dlorenc** [comment]
It's an LTS release, so about as long as anything else I'd guess :)

**Pull request merged**